tests/test_advanced_flag_emission.py

```
@@ -0,0 +1,149 @@
+        plan = _build(tab_key, advanced_state, tmp_path)
+        if not _has_flag(plan.argv, d.flag):
+            missing.append(d.flag)
+    assert not missing, f"{tab_key}: enabled advanced flags not emitted: {missing}"
+
+
+@pytest.mark.parametrize("tab_key", sorted(REGISTRY.tabs))
+def test_secret_advanced_flags_routed_to_env(tab_key, tmp_path):
+    secret_defs = [d for d in REGISTRY.advanced_defs(tab_key) if d.secret_env]
+    if not secret_defs:
+        pytest.skip(f"No secret advanced flags for {tab_key}")
+    token = "secret-value-via-env-only"
```

Ruff (0.16.0)

[error] Possible hardcoded password assigned to: "token"

(S105)